### PR TITLE
[IMP] add tracing

### PR DESCRIPTION
**Pull request by VincentSchippefilt**
## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

image

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -523,23 +530,29 @@ export class Model extends EventBus<any> implements CommandDispatcher {
         }
         this.status = Status.Running;
         const { changes, commands } = this.state.recordChanges(() => {
+          const start = performance.now();
+          console.group(`Dispatching of ${type}`);
```
Comment (LucasLefevre) [inline src/model.ts:534]: I feel like logging every single command is too much.
Something as simple as scrolling spams the console :/

Maybe log it only if it took more than X ms ?

File: src/plugins/ui_core_views/cell_evaluation/evaluator.ts [hunk 2]
```diff
@@ -112,13 +112,15 @@ export class Evaluator {
...
   evaluateCells(positions: CellPosition[]) {
+    const start = performance.now();
     const cellsToCompute = this.createEmptyPositionSet();
     cellsToCompute.addMany(positions);
     const arrayFormulasPositions = this.getArrayFormulasImpactedByChangesOf(positions);
     cellsToCompute.addMany(this.getCellsDependingOn(positions));
     cellsToCompute.addMany(arrayFormulasPositions);
     cellsToCompute.addMany(this.getCellsDependingOn(arrayFormulasPositions));
     this.evaluate(cellsToCompute);
+    console.info("evaluateCells (ms):", performance.now() - start);
```
Comment (LucasLefevre) [inline src/plugins/ui_core_views/cell_evaluation/evaluator.ts:123]: Every other logging is plain English but here it's the implementation name. I'd use plain English everywhere
 ```suggestion
     console.info("evaluate cells (ms):", performance.now() - start);
```

File: src/plugins/ui_core_views/cell_evaluation/evaluator.ts [hunk 3]
```diff
@@ -161,8 +163,10 @@ export class Evaluator {
   }
 
   evaluateAllCells() {
+    const start = performance.now();
     this.evaluatedCells = new PositionMap();
     this.evaluate(this.getAllCells());
+    console.info("evaluateAllCells (ms):", performance.now() - start);
```
Comment (LucasLefevre) [inline src/plugins/ui_core_views/cell_evaluation/evaluator.ts:169]:
```suggestion
    console.info("evaluate all cells (ms):", performance.now() - start);
```

File: src/state_observer.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ export class StateObserver {
   recordChanges(callback: () => void): { changes: HistoryChange[]; commands: CoreCommand[] } {
     this.changes = [];
     this.commands = [];
+
```
Comment (LucasLefevre) [inline src/state_observer.ts:15]: useless diff in untouched file :)

File: src/model.ts [hunk 5]
```diff
@@ -523,23 +530,29 @@ export class Model extends EventBus<any> implements CommandDispatcher {
...
         break;
       case Status.Running:
+        const start = performance.now();
         if (isCoreCommand(command)) {
           const dispatchResult = this.checkDispatchAllowed(command);
           if (!dispatchResult.isSuccessful) {
             return dispatchResult;
           }
           this.state.addCommand(command);
+          console.debug(type, performance.now() - start, "ms");
```
Comment (LucasLefevre) [inline src/model.ts:555]: I would not do that. It logs *every* sub-command dispatched by UI plugins, which might be a looooot (autofilling 10k rows).
Logging is probably not free performance-wise.

**Review by LucasLefevre — COMMENTED** [review]
Some suggestions :)

**Review by pro-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**